feat: release 2.5.0 (#2945)
google/osv-scanner: CHANGELOG.md

# v2.5.0

> ### Features & Refactors:

Reviewer: There are probably more features in the form of merging PRPs from scalibr?

- **Full OSV-Scalibr pipeline**: Migrated scanning, filtering, and matching in `osv-scanner` to use `osv-scalibr` end-to-end, so most plugins that's supported in osv-scalibr should be supported via the `--experimental-plugins` flag ([#2935](https://github.com/google/osv-scanner/pull/2935)).

> -

Reviewer: ```suggestion
- **New extractors and ecosystem support via `osv-scalibr`**:
  - Add `javascript/vsix` extractor to support scanning VS Code extension (`.vsix`) packages.
  - Extend ecosystem mapping for:
    - SUSE
    - Azure Linux / Mariner
    - Alpaquita
    - Mageia
    - openSUSE Leap
    - Debian and Ubuntu PURL
```

- **PURL Type Resolution**: Updated `osvscannerjson` extractor to map ecosystem names to valid PURL types (golang, gem, cargo, npm, etc.).

### Fixes:

- [Bug #2915](https://github.com/google/osv-scanner/pull/2915) Fix issue where osv-scanner reported already-fixed advisories as unfixed for RHEL-family RPM packages (Red Hat, AlmaLinux, Rocky Linux) with epochs by sending epoch-qualified versions.
- Fix Python requirements.txt extractor in `osv-scalibr`, specifically regular expressions used to extract package names and per-requirement options (Fixes #2940, #2931)

> 

Reviewer: ```suggestion
- Fix NPM and Composer PURL generation in `osv-scalibr`, separating package namespace (scope) from package name

```

> 

Reviewer: ```suggestion
- Fix Python requirements.txt extractor in `osv-scalibr`, specifically regular expressions used to extract package names and per-requirement options (Fixes #2940, #2931)
```

# v2.5.0

### Features & Refactors:

> - **End-to-End Scalibr Migration**: Migrated scanning, filtering, and matching in `osv-scanner` to use `osv-scalibr` end-to-end ([#2935](https://github.com/google/osv-scanner/pull/2935)).

Reviewer 2: ```suggestion
- **Full OSV-Scalibr pipeline**: Migrated scanning, filtering, and matching in `osv-scanner` to use `osv-scalibr` end-to-end, so most plugins that's supported in osv-scalibr should be supported via the `--experimental-plugins` flag ([#2935](https://github.com/google/osv-scanner/pull/2935)).
- 

```

>   - **Matching Offloaded to Scalibr Enrichers**: Removed legacy vulnerability and license matchers from `osv-scanner`, delegating them to `osv-scalibr` enrichers.

Reviewer 2: ```suggestion
```

>   - **Filtering Annotator Plugin**: Implemented package filtering as a `scalibr` annotator plugin (`osv-scanner/filter`) to handle unscannable packages, container-specific package exclusions, and ignore rules.

Reviewer 2: ```suggestion
```

>   - **Client Factories**: Introduced `ClientFactories` implementing `scalibrconfig.ClientFactories` to inject User-Agent headers and pool client connections.

Reviewer 2: ```suggestion
```
